Just in case you don't want to mess with the VPN, I've pasted the failing output from the test suite below. FWIW, 'pytest' is a real bucket-type that was created and activated before the suite ran, and should not result in a failure.

``````
======================================================================
ERROR: test_btype_list_buckets (riak.tests.test_all.RiakPbcTransportTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/buildbot/masters/riak/python-client-test/build/riak/tests/test_btypes.py", line 88, in test_btype_list_buckets
    self.assertIn(bucket, btype.get_buckets())
  File "/home/buildbot/masters/riak/python-client-test/build/riak/bucket.py", line 627, in get_buckets
    return self._client.get_buckets(bucket_type=self, timeout=timeout)
  File "/home/buildbot/masters/riak/python-client-test/build/riak/client/transport.py", line 184, in wrapper
    return self._with_retries(pool, thunk)
  File "/home/buildbot/masters/riak/python-client-test/build/riak/client/transport.py", line 126, in _with_retries
    return fn(transport)
  File "/home/buildbot/masters/riak/python-client-test/build/riak/client/transport.py", line 182, in thunk
    return fn(self, transport, *args, **kwargs)
  File "/home/buildbot/masters/riak/python-client-test/build/riak/client/operations.py", line 62, in get_buckets
    timeout=timeout)]
  File "/home/buildbot/masters/riak/python-client-test/build/riak/transports/pbc/transport.py", line 292, in get_buckets
    MSG_CODE_LIST_BUCKETS_RESP)
  File "/home/buildbot/masters/riak/python-client-test/build/riak/transports/pbc/connection.py", line 50, in _request
    return self._recv_msg(expect)
  File "/home/buildbot/masters/riak/python-client-test/build/riak/transports/pbc/connection.py", line 143, in _recv_msg
    raise RiakError(err.errmsg)
RiakError: "No bucket-type named 'pytest'"

======================================================================
ERROR: test_default_btype_list_buckets (riak.tests.test_all.RiakPbcTransportTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/buildbot/masters/riak/python-client-test/build/riak/tests/test_btypes.py", line 119, in test_default_btype_list_buckets
    self.assertIn(bucket, default_btype.get_buckets())
  File "/home/buildbot/masters/riak/python-client-test/build/riak/bucket.py", line 627, in get_buckets
    return self._client.get_buckets(bucket_type=self, timeout=timeout)
  File "/home/buildbot/masters/riak/python-client-test/build/riak/client/transport.py", line 184, in wrapper
    return self._with_retries(pool, thunk)
  File "/home/buildbot/masters/riak/python-client-test/build/riak/client/transport.py", line 126, in _with_retries
    return fn(transport)
  File "/home/buildbot/masters/riak/python-client-test/build/riak/client/transport.py", line 182, in thunk
    return fn(self, transport, *args, **kwargs)
  File "/home/buildbot/masters/riak/python-client-test/build/riak/client/operations.py", line 62, in get_buckets
    timeout=timeout)]
  File "/home/buildbot/masters/riak/python-client-test/build/riak/transports/pbc/transport.py", line 292, in get_buckets
    MSG_CODE_LIST_BUCKETS_RESP)
  File "/home/buildbot/masters/riak/python-client-test/build/riak/transports/pbc/connection.py", line 50, in _request
    return self._recv_msg(expect)
  File "/home/buildbot/masters/riak/python-client-test/build/riak/transports/pbc/connection.py", line 143, in _recv_msg
    raise RiakError(err.errmsg)
RiakError: "No bucket-type named 'undefined'"

======================================================================
ERROR: test_list_buckets (riak.tests.test_all.RiakPbcTransportTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/buildbot/masters/riak/python-client-test/build/riak/tests/test_kv.py", line 454, in test_list_buckets
    buckets = self.client.get_buckets()
  File "/home/buildbot/masters/riak/python-client-test/build/riak/client/transport.py", line 184, in wrapper
    return self._with_retries(pool, thunk)
  File "/home/buildbot/masters/riak/python-client-test/build/riak/client/transport.py", line 126, in _with_retries
    return fn(transport)
  File "/home/buildbot/masters/riak/python-client-test/build/riak/client/transport.py", line 182, in thunk
    return fn(self, transport, *args, **kwargs)
  File "/home/buildbot/masters/riak/python-client-test/build/riak/client/operations.py", line 62, in get_buckets
    timeout=timeout)]
  File "/home/buildbot/masters/riak/python-client-test/build/riak/transports/pbc/transport.py", line 292, in get_buckets
    MSG_CODE_LIST_BUCKETS_RESP)
  File "/home/buildbot/masters/riak/python-client-test/build/riak/transports/pbc/connection.py", line 50, in _request
    return self._recv_msg(expect)
  File "/home/buildbot/masters/riak/python-client-test/build/riak/transports/pbc/connection.py", line 143, in _recv_msg
    raise RiakError(err.errmsg)
RiakError: "No bucket-type named 'undefined'"

======================================================================
ERROR: test_stream_buckets (riak.tests.test_all.RiakPbcTransportTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/buildbot/masters/riak/python-client-test/build/riak/tests/test_kv.py", line 462, in test_stream_buckets
    for bucket_list in self.client.stream_buckets():
  File "/home/buildbot/masters/riak/python-client-test/build/riak/client/operations.py", line 89, in stream_buckets
    for bucket_list in stream:
  File "/home/buildbot/masters/riak/python-client-test/build/riak/transports/pbc/stream.py", line 118, in next
    response = super(RiakPbcBucketStream, self).next()
  File "/home/buildbot/masters/riak/python-client-test/build/riak/transports/pbc/stream.py", line 51, in next
    msg_code, resp = self.transport._recv_msg(expect=self._expect)
  File "/home/buildbot/masters/riak/python-client-test/build/riak/transports/pbc/connection.py", line 143, in _recv_msg
    raise RiakError(err.errmsg)
RiakError: "No bucket-type named 'undefined'"

======================================================================
ERROR: test_stream_buckets_abort (riak.tests.test_all.RiakPbcTransportTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/buildbot/masters/riak/python-client-test/build/riak/tests/test_kv.py", line 472, in test_stream_buckets_abort
    for bucket_list in self.client.stream_buckets():
  File "/home/buildbot/masters/riak/python-client-test/build/riak/client/operations.py", line 89, in stream_buckets
    for bucket_list in stream:
  File "/home/buildbot/masters/riak/python-client-test/build/riak/transports/pbc/stream.py", line 118, in next
    response = super(RiakPbcBucketStream, self).next()
  File "/home/buildbot/masters/riak/python-client-test/build/riak/transports/pbc/stream.py", line 51, in next
    msg_code, resp = self.transport._recv_msg(expect=self._expect)
  File "/home/buildbot/masters/riak/python-client-test/build/riak/transports/pbc/connection.py", line 143, in _recv_msg
    raise RiakError(err.errmsg)
RiakError: "No bucket-type named 'undefined'"

----------------------------------------------------------------------```
``````

--- a/src/riak_kv_pb_bucket.erl
+++ b/src/riak_kv_pb_bucket.erl
@@ -41,6 +41,7 @@
 
 -module(riak_kv_pb_bucket).
 
+-type optional(A) :: A | undefined.
 -include_lib("riak_pb/include/riak_kv_pb.hrl").
 
 -behaviour(riak_api_pb_service).
@@ -160,6 +161,9 @@ process_stream({ReqId, Error}, ReqId,
     {error, {format, Error}, State#state{req = undefined, req_ctx = undefined}}.
 
 
+-spec do_list_buckets(binary(), optional(pos_integer()), optional(boolean()), #rpblistbucketsreq{}, #state{}) ->
+                             {reply, tuple(), #state{}} |
+                             {error, {format, iodata()}, #state{}}.
 do_list_buckets(Type, Timeout, true, Req, #state{client=C}=State) ->
     {ok, ReqId} = C:stream_list_buckets(none, Timeout, Type),
     {reply, {stream, ReqId}, State#state{req = Req, req_ctx = ReqId}};
@@ -171,6 +175,7 @@ do_list_buckets(Type, Timeout, _Stream, _Req, #state{client=C}=State) ->
             {error, {format, Reason}, State}
     end.
 
+-spec check_bucket_type(optional(binary())) -> {ok, binary()} | error.
 check_bucket_type(undefined) -> {ok, <<"default">>};
 check_bucket_type(<<"default">>) -> {ok, <<"default">>};
 check_bucket_type(Type) ->
@@ -181,6 +186,7 @@ check_bucket_type(Type) ->
             error
     end.
 
+-spec maybe_create_bucket_type(binary(), binary()) -> binary() | {binary(), binary()}.
 maybe_create_bucket_type(<<"default">>, Bucket) ->
     Bucket;
 maybe_create_bucket_type(Type, Bucket) when is_binary(Type) ->

